cmd/podman/common/create_opts.go

```diff
@@ -517,7 +517,12 @@ func ContainerCreateToContainerCLIOpts(cc handlers.CreateContainerConfig, rtc *c
 		cliOpts.OOMKillDisable = *cc.HostConfig.OomKillDisable
 	}
 	if cc.Config.Healthcheck != nil {
-		cliOpts.HealthCmd = strings.Join(cc.Config.Healthcheck.Test, " ")
+		finCmd := ""
+		for _, str := range cc.Config.Healthcheck.Test {
+			finCmd = finCmd + str + " "
+		}
+		finCmd = finCmd[:len(finCmd)-1]
```

Comment:
Don't think this is safe for all commands - Manpages say that a length-1 command is allowed to be specified without brackets, so `podman run --health-cmd "./my_script,sh"` is perfectly legal, and will not work with this.

Reply:
fixed